src/google-classroom-extractor/edfi_google_classroom_extractor/mapping/assignment_submissions.py

```diff
@@ -63,6 +66,35 @@ def derive_state(submission_row: Series) -> str:
     return api_state
 
 
+def _get_submission_datetime(submission_row: Series) -> str:
+    if submission_row["state"] == TURNED_IN_STATE:
+        return submission_row["updateTime"]
+
+    status_history: List[dict] = ast.literal_eval(submission_row["submissionHistory"])
+    # this submission history also has a history of grades, but for this purpose, we only care
+    # about stateHistory so we filter by stateHistory
+    status_history = [
+        status_item["stateHistory"]
+        for status_item in status_history
+        if "stateHistory" in status_item.keys()
+    ]
+    status_history = sorted(
+        status_history, key=lambda k: k["stateTimestamp"], reverse=True
+    )
+
+    for status_record in status_history:
+        status_is_turned_in: bool = status_record["state"] == TURNED_IN_STATE
+        status_is_reclaimed_by_student: bool = status_record["state"] == RECLAIMED_STATE
```

Comment:
You could simplify this algorithm at the top: if the `submission_row["state"]` is not `TURNED_IN` or `RETURNED`, then just return None. That would allow us to skip several lines of code when the submission is in any other status.

Reply:
✅